build.assets/arch.mk

```diff
@@ -1,9 +1,14 @@
 # These variables are extracted from build.assets/Makefile so they can be imported
 # by other Makefiles
-ifeq ($(origin ARCH), undefined) # avoid use of "ARCH ?= $(...)" as the lazy loading will repeatedly re-run the same shell command
-	ARCH := $(shell go env GOARCH)
-endif
+
+# Eval $(ARCH) one time max, or not at all if $(ARCH) is not used.
+# Calling `go` can emit an error if not installed, which we do not
+# want to do if $(ARCH) is never used.
+# https://make.mad-scientist.net/deferred-simple-variable-expansion/
```

Comment:
Nice! Does this also fix #37784?

Reply:
Yes and no. I will look at fixing that soon. I was hesitant about doing this before because it really is a bit of gobbledegook but I noticed that even when Go is installed, if it is the wrong version according to the `toolchain` directive in `go.mod`, it will now download the newer toolchain if an older one is installed. And it does this even if it never uses Go because of the make vars.

But the reason it occurs in the main `Makefile` is because it has a bare `export` directive which causes all make vars to be evaluated whether they are used or not so it can export them to the environment. This trick for deferred simple variable expansion wont work on the main `Makefile` until we get rid of that bare `export`. I do want to do that but it does require a bit of an audit to understand just what does need to be exported and export only those vars.